src/common/connection.ts

```diff
@@ -15,12 +13,49 @@ function isStreamMessageType(type) {
          type === 'path_find'
 }
 
-class Connection extends events.EventEmitter {
-  constructor(url, options = {}) {
+interface ConnectionOptions {
+  trace?: boolean,
+  proxy?: string
+  proxyAuthorization?: string
+  authorization?: string
+  trustedCertificates?: string[]
+  key?: string
+  passphrase?: string
+  certificate?: string
+  timeout?: number
+}
+
+class Connection extends EventEmitter {
+
+  private _url: string
+  private _trace: boolean
+  private _console?: Console
+  private _proxyURL?: string
+  private _proxyAuthorization?: string
+  private _authorization?: string
+  private _trustedCertificates?: string[]
+  // TODO: options.key never used?
```

Comment:
Looks like it's used in `_createWebSocket()`

Reply:
Ha, can't believe I missed that